app/models/game.rb

```diff
@@ -51,4 +54,10 @@ def populate_board
   def self.available
     where('white_player_id IS NULL OR black_player_id IS NULL')
   end
+
+  def forfeit(current_user)
+    update_attributes(winner_id: white_player_id, result: 'Forfeit', black_player_id: nil) if current_user.id == black_player_id
+
+    update_attributes(winner_id: black_player_id, result: 'Forfeit', white_player_id: nil) if current_user.id == white_player_id
+  end
```

Comment:
This looks good.

One change: remove `black_player_id: nil` and `white_player_id: nil`.

A player can be associated as both a black/white player and the winner at the same time.

Reply:
`current_user.id == white_player_id`

Could be `current_user == white_player`